################################################################################
SHORT DESCRIPTION:
################################################################################
Handle the xci cpuid signature in LibXL

################################################################################
LONG DESCRIPTION:
################################################################################
This is a hack from a long time ago, when we had to use custom pv frontends.
I is now disabled by default but still very useful for debugging.
A seperate patch introduces the feature as a hypercall, this one just adds
support for it in XL VM configuration files.

################################################################################
CHANGELOG
################################################################################
Authors:
Jed Lejosne <lejosnej@ainfosec.com>

################################################################################
REMOVAL
################################################################################
Removing this won't break anything, but disable a useful debugging feature.

################################################################################
UPSTREAM PLAN
################################################################################
This is specific to OpenXT, nothing to upstream here.

################################################################################
INTERNAL DEPENDENCIES
################################################################################
hvm-cpuid-multicore.patch

################################################################################
PATCHES
################################################################################
Index: xen-4.6.4/tools/libxl/libxl_create.c
===================================================================
--- xen-4.6.4.orig/tools/libxl/libxl_create.c
+++ xen-4.6.4/tools/libxl/libxl_create.c
@@ -312,6 +312,7 @@ int libxl__domain_build_info_setdefault(
         libxl_defbool_setdefault(&b_info->u.hvm.vpt_align,          true);
         libxl_defbool_setdefault(&b_info->u.hvm.nested_hvm,         false);
         libxl_defbool_setdefault(&b_info->u.hvm.altp2m,             false);
+        libxl_defbool_setdefault(&b_info->u.hvm.xci_cpuid_signature, false);
         libxl_defbool_setdefault(&b_info->u.hvm.usb,                false);
         libxl_defbool_setdefault(&b_info->u.hvm.xen_platform_pci,   true);
 
Index: xen-4.6.4/tools/libxl/libxl_dom.c
===================================================================
--- xen-4.6.4.orig/tools/libxl/libxl_dom.c
+++ xen-4.6.4/tools/libxl/libxl_dom.c
@@ -287,6 +287,8 @@ static void hvm_set_conf_params(xc_inter
                     libxl_defbool_val(info->u.hvm.nested_hvm));
     xc_hvm_param_set(handle, domid, HVM_PARAM_ALTP2M,
                     libxl_defbool_val(info->u.hvm.altp2m));
+    xc_hvm_param_set(handle, domid, HVM_PARAM_XCI_CPUID_SIGNATURE,
+                    libxl_defbool_val(info->u.hvm.xci_cpuid_signature));
 }
 
 int libxl__build_pre(libxl__gc *gc, uint32_t domid,
Index: xen-4.6.4/tools/libxl/libxl_types.idl
===================================================================
--- xen-4.6.4.orig/tools/libxl/libxl_types.idl
+++ xen-4.6.4/tools/libxl/libxl_types.idl
@@ -482,6 +482,7 @@ libxl_domain_build_info = Struct("domain
                                        ("timer_mode",       libxl_timer_mode),
                                        ("nested_hvm",       libxl_defbool),
                                        ("altp2m",           libxl_defbool),
+                                       ("xci_cpuid_signature", libxl_defbool),
                                        ("smbios_firmware",  string),
                                        ("acpi_firmware",    string),
                                        ("hdtype",           libxl_hdtype),
Index: xen-4.6.4/tools/libxl/xl_cmdimpl.c
===================================================================
--- xen-4.6.4.orig/tools/libxl/xl_cmdimpl.c
+++ xen-4.6.4/tools/libxl/xl_cmdimpl.c
@@ -1605,6 +1605,8 @@ static void parse_config_data(const char
 
         xlu_cfg_get_defbool(config, "altp2mhvm", &b_info->u.hvm.altp2m, 0);
 
+        xlu_cfg_get_defbool(config, "xci_cpuid_signature", &b_info->u.hvm.xci_cpuid_signature, 0);
+
         xlu_cfg_replace_string(config, "smbios_firmware",
                                &b_info->u.hvm.smbios_firmware, 0);
         xlu_cfg_replace_string(config, "acpi_firmware",
